packages/frontend/src/components/staking/components/StakingAction.js

```diff
@@ -47,9 +54,47 @@ export default function StakingAction({
     const displayAmount = useMax ? formatNearAmount(amount, 5).replace(/,/g, '') : amount;
     const availableToStake = availableBalance;
     const invalidStakeActionAmount = new BN(useMax ? amount : parseNearAmount(amount)).sub(new BN(stake ? availableToStake : staked)).gt(new BN(STAKING_AMOUNT_DEVIATION)) || !isDecimalString(amount);
-    const stakeActionAllowed = hasStakeActionAmount && !invalidStakeActionAmount && !success;
+    const stakeActionAllowed = hasStakeActionAmount && !invalidStakeActionAmount && !success && !disableStaking;
     const stakeNotAllowed = !!selectedValidator && selectedValidator !== match.params.validator && !!currentValidators.length;
 
+    const validatorHasFAK = async (validator) => {
+        const accessKeys = await wallet.getAccessKeys(validator);
+        const fullAccessKeys = accessKeys.filter((key) => (
```

Comment:
If `filter` is changed to `some` then this expression can be returned directly

Reply:
Good point. Done.